File: 3191_minimum_operations_to_make_binary_array_elements_equal_to_one_i.py
class Solution(object):
    def minOperations(self, r):
        x = 0
        for i in range(len(r) - 2):
            if not r[i]: r[i], r[i + 1], r[i + 2], x = 1 - r[i], 1 - r[i + 1], 1 - r[i + 2], x + 1
        return -1 if any(v == 0 for v in r) else x

File: 3191_minimum_operations_to_make_binary_array_elements_equal_to_one_i.cpp
class Solution {
public:
    int minOperations(vector<int>& z) {
        int x = 0, n = z.size();
        for (int i = 0; i < n - 2; i++)
            if (!z[i]) z[i] ^= 1, z[i + 1] ^= 1, z[i + 2] ^= 1, x++;
        return count(z.begin(), z.end(), 0) ? -1 : x;
    }
};
